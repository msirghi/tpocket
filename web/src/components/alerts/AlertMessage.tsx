import React from 'react';
import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
import Alert from '@material-ui/lab/Alert';
import { AlertType } from '../../commons/enums';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: '100%',
      '& > * + *': {
        marginTop: theme.spacing(2)
      }
    }
  })
);

type Props = {
  type: AlertType;
  message: string;
};

export const AlertMessage: React.FC<Props> = ({ message, type }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Alert  severity={type} color={type}>
        {message}
      </Alert>
    </div>
  );
};
