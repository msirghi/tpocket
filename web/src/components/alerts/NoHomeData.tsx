import React from 'react';
import Typography from '@material-ui/core/Typography';

export const NoHomeData: React.FC = () => {
  return (
    <div className='no-data-wrapper'>
      <img
        height={200}
        src='https://www.pinclipart.com/picdir/big/220-2203662_man-with-smartphone-illustration-smartphone-person-dribbble-clipart.png'
        alt='No data'
      />
      <div className='mt-5rem text-center'>
        <Typography gutterBottom variant='h6' component='h5'>
          Look like you don't have any data.
        </Typography>
        <Typography gutterBottom variant='body1' component='h5'>
          Try adding categories & expenses to see the statistics.
        </Typography>
      </div>
    </div>
  );
};
