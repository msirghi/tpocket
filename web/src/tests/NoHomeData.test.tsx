import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { NoHomeData } from '../components/alerts/NoHomeData';

describe('NoHomeData component', () => {
  it('should render properly', () => {
    const component = shallow(<NoHomeData />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
