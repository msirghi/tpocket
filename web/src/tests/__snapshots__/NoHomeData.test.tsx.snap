// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoHomeData component should render properly 1`] = `
<div
  className="no-data-wrapper"
>
  <img
    alt="No data"
    height={200}
    src="https://www.pinclipart.com/picdir/big/220-2203662_man-with-smartphone-illustration-smartphone-person-dribbble-clipart.png"
  />
  <div
    className="mt-5rem text-center"
  >
    <WithStyles(ForwardRef(Typography))
      component="h5"
      gutterBottom={true}
      variant="h6"
    >
      Look like you don't have any data.
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      component="h5"
      gutterBottom={true}
      variant="body1"
    >
      Try adding categories & expenses to see the statistics.
    </WithStyles(ForwardRef(Typography))>
  </div>
</div>
`;
